### PR TITLE
Fixed preprocessor inheritance to not monkey patch base methods

### PR DESCRIPTION
**Pull request by MSeal**
Addresses https://github.com/nteract/papermill/issues/129

Test added fails without the change removing the monkey patching.

Extra preprocess code that is copied over can be removed once https://github.com/jupyter/nbconvert/pull/811 is merged.

### CODE REVIEW

File: papermill/preprocess.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            Additional resources used in the conversion process.
+        """
+        path = resources.get('metadata', {}).get('path', '')
+        if path == '':
+            path = None
+
+        # clear display_id map
+        self._display_id_map = {}
+
+        # from jupyter_client.manager import start_new_kernel
```
Comment (betatim) [inline papermill/preprocess.py]: delete unused import?

File: papermill/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+        Returns
+        -------
+        nb : NotebookNode
+            The executed notebook.
+        resources : dictionary
+            Additional resources used in the conversion process.
+        """
+        path = resources.get('metadata', {}).get('path', '')
+        if path == '':
```
Comment (betatim) [inline papermill/preprocess.py]: for more pythonicness: `if not path`.

File: papermill/preprocess.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        self.kc.allow_stdin = False
+        self.nb = nb
+
+        try:
+            nb, resources = self.papermill_preprocess(nb, resources)
+        finally:
+            self.kc.stop_channels()
+            self.km.shutdown_kernel(now=self.shutdown_kernel == 'immediate')
+
+        delattr(self, 'nb')
```
Comment (betatim) [inline papermill/preprocess.py]: Using `delattr` makes me take note as it is so rare. Should there be a comment explaining why we assign it on line 90 and then get rid of it again here. Seemingly never having done anything with it?

Comment (MSeal) [reply papermill/preprocess.py]: That function was moved without any changes. I can clean it up in the move process.

Comment (betatim) [reply papermill/preprocess.py]: Thanks! Tidying as we go along :)

File: papermill/preprocess.py (hunk 2 above)
Comment (mpacer) [reply papermill/preprocess.py]: or if you want it to be strict, `if path is ''` since strings are immutable.

Comment (betatim) [reply papermill/preprocess.py]: PEP8 would like you to make use of the falsey-ness of empty strings https://www.python.org/dev/peps/pep-0008/#programming-recommendations not that I am a orthodox PEP8 follower or anything *coughcough* ;)

Comment (rgbkrk) [reply papermill/preprocess.py]: We should probably pick one path, so I'll line up with the PEP8 recommendation of `if not path`.

**Comment by MSeal** [comment]
I should be able to apply basic cleanups on code I touched this week, been catching up on tasks from being sick.

File: papermill/preprocess.py (hunk 2 above)
Comment (MSeal) [reply papermill/preprocess.py]: I'm just removing the if statement `path = resources.get('metadata', {}).get('path') or None`

**Pull request merged**